src/distributions/utils.rs

```diff
@@ -138,6 +138,10 @@ macro_rules! wmul_impl_usize {
         }
     };
 }
+#[cfg(target_pointer_width = "8")]
+wmul_impl_usize! { u8 }
```

Comment:
Note that Rust cannot have 8-bit wide pointers, so this `u8` code is unnecessary. I can confirm that `rand` builds for `avr-unknown-gnu-atmega328` with only the `u16` lines.

See https://github.com/rust-lang/rust/pull/49305 for more information. Basically, Rust `usize` must be at least 16-bits because, C99 says the corresponding C type (`uintptr_t`) is at least 16-bits.

Reply:
Oh yep, my mixup